### PR TITLE
Add support for private video

### LINKED ISSUE #7: Enable private video upload & management
Currently it is not possible to upload videos as private from the Strapi plugin. As a user with a requirement to keep video secure at all times we need this to be possible.

Add a toggle for private video to the Strapi upload page. We should be able to then watch and manage private video using the api key for auth. This should be handled behind the scenes.

### PR DESCRIPTION
**Pull request by Tommo56700**
Adds a toggle for private video - default can be set from config
Resolves #7

### CODE REVIEW

File: src/admin/components/Modal/updateVideo/PlayerView.tsx [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
     const { videoId } = video
 
+    const getToken = async () => {
+        const tokenResponse = await assetsRequests.getToken(videoId)
+        setToken(tokenResponse.token)
+        setIsGettingToken(false)
+    }
+
+    useEffect(() => {
+        getToken()
```
Comment (olivierapivideo) [inline src/admin/components/Modal/updateVideo/PlayerView.tsx:26]: I think this should only be done when the video is private

File: src/admin/components/Videos/index.tsx [hunk 2]
```diff
@@ -31,10 +35,26 @@ const VideoView: FC<IVideosProps> = ({ video, updateData, deletable, editable })
...
     const formatedCreatedAt = getDayMonthYearHourDate(createdAt)
 
+    const getToken = async () => {
+        const tokenResponse = await assetsRequests.getToken(videoId)
+        setToken(tokenResponse.token)
+        setIsGettingToken(false)
+    }
+
+    useEffect(() => {
+        getToken()
```
Comment (olivierapivideo) [inline src/admin/components/Videos/index.tsx:45]: I think this should only be done when the video is private

File: src/admin/pages/Settings/index.tsx [hunk 3]
```diff
@@ -85,14 +95,24 @@ const Settings = () => {
...
-                                    value={apiKey}
+                                    value={settings.apiKey}
                                     placeholder="Enter your API Key"
                                     description="Generated in the api.video's dashboard and used for authenticating API calls."
                                     detailsLink="https://dashboard.api.video"
                                     isPassword
                                     onChange={handleChange}
                                 />
                             </GridItem>
+                            <GridItem col={6} s={12}>
```
Comment (olivierapivideo) [inline src/admin/pages/Settings/index.tsx:106]: It's a detail, but the 2 fields are not quite aligned at my place

File: src/server/services/settings-services.ts [hunk 4]
```diff
@@ -1,34 +1,38 @@
...
             environment: strapi.config.environment,
             type: 'plugin',
             name: 'api-video-uploader',
         })
 
-        const configKey = await pluginStore.get({
-            key: 'apiKey',
+        const customSettings = await pluginStore.get({
+            key: 'settings',
         })
```
Comment (olivierapivideo) [inline src/server/services/settings-services.ts:15]: I'm afraid it's not backwards compatible. For users of the current version, the api key they are currently using will be lost, right? Maybe by using 2 keys (the old "apiKey" + a new one for the default value of private) we wouldn't have this problem ?

**Review by olivierapivideo — CHANGES_REQUESTED** [review]
Hi Tommo!
Thanks for this great contribution!

I have added some comments in the review.

On the other hand, I have trouble seeing how the plugin can be used via the Strapi user API. 
As it stands, when the user retrieves private videos by calling `GET /api/api-video-uploader/api-video-asset`, it will return the assets from the Strapi database, with tokens that will never be regenerated.  Have you thought of a solution to this problem? Or maybe for the way you use the plugin this is not a problem?

File: src/server/services/settings-services.ts (hunk 4 above)
Comment (Tommo56700) [reply src/server/services/settings-services.ts:15]: This is true API keys will be lost. I prefer the separation - although there is then a question of how to handle the saving of the settings page if one setting succeeds and the other fails.

**Comment by Tommo56700** [comment]
To preface this, I have little experience with designing strapi plugins.

Yes, there is indeed the unsolved problem of the static token in the Strapi database. In my opinion it makes little sense to store the token database whatsoever. Do you know of a way to "hook" into the controller of `GET /api/api-video-uploader/api-video-asset` before it is sent over the api? If so, a token could be fetched at this point and attached to the response. Any thoughts?

**Comment by olivierapivideo** [comment]
Hey @Tommo56700 , if that's fine for you I can implement this missing feature
I'm going to have a look on how to do that

**Pull request closed without merging**